test/functional/ui/popupmenu_spec.lua

```diff
@@ -2238,6 +2238,33 @@ describe('builtin popupmenu', function()
         feed('S<C-x><C-o><C-N>')
         eq(1, n.eval([[len(uniq(copy(g:bufnrs))) == 1]]))
       end)
+
+      it('handles tabs in info width calculation', function()
+        screen:try_resize(50, 11)
+        command([[
+          set cot+=menuone
+          let g:list = [#{word: 'class', info: "\tClassName() = default;"}]
+        ]])
+        feed('S<C-x><C-o>')
+        local info = fn.complete_info()
+        eq(30, api.nvim_win_get_width(info.preview_winid))
+        feed('<ESC>')
+        exec([[
+          setlocal tabstop=1
+          autocmd ModeChanged *:i ++once call complete(1, [#{word: 'a'}])
+                \| call nvim__complete_set(0, #{info: "\tfloob\tfloob"})
+        ]])
+        feed('i')
+        info = fn.complete_info()
+        eq(12, api.nvim_win_get_width(info.preview_winid))
```

Comment:
```suggestion
        eq(21, api.nvim_win_get_width(info.preview_winid))
```

After making the other change, this should be 21. (default of `&ts=8` aligns the end of the 2nd tab to column 16)

Might be worth also adding a screen test here to ensure the text isn't being wrapped?